From 2f470ed06481099b2e6d2c249a94b292dcc562ed Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Mon, 25 Jul 2022 11:14:59 +0200
Subject: [PATCH] Revert "x11/input-selection-stream: Handle Xwayland going
 away"

The backport is for a fix that isn't applicable to the gnome-41 stable
branch, since it relies on selections themselves owning the X11 window.
The fix was about cleaning up that window on finalize, but that falls
apart of the window isn't actually owned by the selection itself.

This reverts commit 16b7dce9cb1313d08a594c20697cea9d0b8812f0.
---
 src/x11/meta-x11-selection-input-stream.c | 24 ++---------------------
 1 file changed, 2 insertions(+), 22 deletions(-)

diff --git a/src/x11/meta-x11-selection-input-stream.c b/src/x11/meta-x11-selection-input-stream.c
index 7ed64ac0b5..e8d7820478 100644
--- a/src/x11/meta-x11-selection-input-stream.c
+++ b/src/x11/meta-x11-selection-input-stream.c
@@ -268,14 +268,9 @@ meta_x11_selection_input_stream_dispose (GObject *object)
     META_X11_SELECTION_INPUT_STREAM (object);
   MetaX11SelectionInputStreamPrivate *priv =
     meta_x11_selection_input_stream_get_instance_private (stream);
-  MetaX11Display *x11_display;
 
-  x11_display = priv->x11_display;
-  if (x11_display)
-    {
-      x11_display->selection.input_streams =
-        g_list_remove (x11_display->selection.input_streams, stream);
-    }
+  priv->x11_display->selection.input_streams =
+    g_list_remove (priv->x11_display->selection.input_streams, stream);
 
   G_OBJECT_CLASS (meta_x11_selection_input_stream_parent_class)->dispose (object);
 }
@@ -287,7 +282,6 @@ meta_x11_selection_input_stream_finalize (GObject *object)
     META_X11_SELECTION_INPUT_STREAM (object);
   MetaX11SelectionInputStreamPrivate *priv =
     meta_x11_selection_input_stream_get_instance_private (stream);
-  MetaX11Display *x11_display;
 
   g_async_queue_unref (priv->chunks);
 
@@ -295,17 +289,6 @@ meta_x11_selection_input_stream_finalize (GObject *object)
   g_free (priv->target);
   g_free (priv->property);
 
-  x11_display = priv->x11_display;
-  if (x11_display)
-    {
-      Display *xdisplay = meta_x11_display_get_xdisplay (x11_display);
-
-      XDestroyWindow (xdisplay, priv->window);
-
-      g_object_remove_weak_pointer (G_OBJECT (x11_display),
-                                    (gpointer *) &priv->x11_display);
-    }
-
   G_OBJECT_CLASS (meta_x11_selection_input_stream_parent_class)->finalize (object);
 }
 
@@ -537,9 +520,6 @@ meta_x11_selection_input_stream_new_async (MetaX11Display      *x11_display,
   priv = meta_x11_selection_input_stream_get_instance_private (stream);
 
   priv->x11_display = x11_display;
-  g_object_add_weak_pointer (G_OBJECT (x11_display),
-                             (gpointer *) &priv->x11_display);
-
   x11_display->selection.input_streams =
     g_list_prepend (x11_display->selection.input_streams, stream);
   priv->selection = g_strdup (selection);
-- 
2.35.3

